tools/msys2_fetch_and_build_all_tx16s.sh

```diff
@@ -15,81 +41,109 @@ do
 	fi
 done
 
+STEP=1
 echo "=== Step $((STEP++)): Creating a directory for EdgeTX ==="
-mkdir ~/edgetx && cd ~/edgetx
+mkdir -p ~/edgetx && cd ~/edgetx
+check_command $? "mkdir -p ~/edgetx && cd ~/edgetx"
 if [[ $PAUSEAFTEREACHLINE == "true" ]]; then
   echo "Step finished. Please check the output above and press Enter to continue or Ctrl+C to stop."
   read
 fi
 
-echo "=== Step $((STEP++)): Fetching EdgeTX source tree (main branch) from GitHub ==="
-git clone --recursive -b main https://github.com/EdgeTX/edgetx.git edgetx_main
+
+if [ ! -d "${SOURCE_DIR}" ]; then
+  echo "=== Step $((STEP++)): Fetching EdgeTX source tree (${BRANCH_NAME} branch) from GitHub ==="
+  git clone --recursive -b ${BRANCH_NAME} https://github.com/EdgeTX/edgetx.git ${SOURCE_DIR}
+  check_command $? "git clone --recursive -b ${BRANCH_NAME} https://github.com/EdgeTX/edgetx.git ${SOURCE_DIR}"
+  cd ${SOURCE_DIR}
+else
+  echo "=== Step $((STEP++)): updating EdgeTX source tree (${BRANCH_NAME} branch) from GitHub ==="
+  cd ${SOURCE_DIR}
+  check_command $? "cd ${SOURCE_DIR}"
+  git checkout ${BRANCH_NAME}
+  check_command $? "git checkout ${BRANCH_NAME}"
+  git pull
```

Comment:
Might be good idea to update the submodules here explicitly as well, e.g. with:
`git submodule update --init --recursive`

Reply:
good idea
done